### PR TITLE
Fix all raised bugs and add coverage support

### LINKED ISSUE #11: Unparseable segment messages are not correctly decoded
http://sqlfluff-online.herokuapp.com/fluffed?sql=H4sIALNhb18C_ytOzUlNLlEw5OVKK8rPVSjh5Uovyi8tUEiqBInxcjmCAS8XAA6-IFMoAAAA&dialect=ansi

The description reads 

```
Found unparsable section: '\n\nAAAAAA'
```

When it should be 

```
Found unparsable section: '

AAAAAA'

```

### LINKED ISSUE #12: I want to submit the form via command-enter
And right now i need to click the button.

### LINKED ISSUE #22: Test for perfect sql is unchanged, has no errors.
I almost merged a PR in which even perfect SQL had a no-newline-at-end-of-file error. That should have been caught in a test.

### LINKED ISSUE #25: Internal error
http://online.sqlfluff.com/fluffed?sql=H4sIAP146GAC_22PsU7DMBCG90h5h1OXtCgKArYiBoYKsSAE7NbVOWKL2I7sS0JU9d2JG1BC1Zvu_u8b7n9-ed-9fYC27OBwgAY9mlAEqcggHI9FiboeRCDfaUnCEHstA6zTBMYJVJPkaT_drvWjxUND-Zx2OmgWJfIyzJQzJIL0RFZ0mvoMMAB1ZFlYNEtVutby-mozC6dkEj69M1A5V9Uk0GI98PhfoTSHifeKPF36BR4gGwuXYWyZbbcxmjW05bJMdHvaZ_-FVUSryF4fn3ZnsMGKRIOsIr8-g7-1Y8-IAzv59V1IZ_7Eyru2gf0ANznc5nCXJpv7NPkBJJr5zKwBAAA%3D&dialect=postgres

### PR DESCRIPTION
**Pull request by tunetheweb**
fixes #12 
fixes #11 
fixes #22 
fixes #25

These are all quite small so have lumped them into one pull request.

I also added coverage and some more tests to ensure we have 100% python coverage

### CODE REVIEW

**Comment by nolanbconaway** [comment]
This looks p good but I won't be able to look closely until later this week!

File: src/app/routes.py [hunk 1]
```diff
@@ -45,12 +45,17 @@ def fluff_results():
 
     dialect = request.args["dialect"]
     linted = lint(sql, dialect=dialect)
+    fixed_sql = ""
+
+    # Only fix if linting isn't completely broken
+    if "Unrecoverable failure" not in str(linted):
```
Comment (barrywhart) [inline src/app/routes.py:51]: Is there a risk that the user's actual SQL input contains the sentinel string, `Unrecoverable failure`? Is there potentially a more precise way to check for failure, such as catching an exception, etc.?

File: test/test_app.py [hunk 2]
```diff
@@ -58,3 +58,22 @@ def test_carriage_return_sql(client):
...
+    sql_encoded = sql_encode(
+        "INSERT into {{ params.schema }}.daily_service_metrics (\nSELECT\n*\nFROM table\n)"
+    )
+    rv = client.get("/fluffed", query_string=f"""dialect=ansi&sql={sql_encoded}""")
+
+    html = rv.data.decode().lower()
+    soup = BeautifulSoup(html, "html.parser")
+    fixed_sql = soup.find("textarea", {"id": "fixedsql"}).text
+
+    # we should get an extra z in there if the carriage returns are not well handled.
```
Comment (barrywhart) [inline test/test_app.py:78]: It looks like this code was copied from the test above, which tests a query whose final character is a `z`. I don't _think_ it's needed in this test.

File: src/app/routes.py (hunk 1 above)
Comment (tunetheweb) [reply src/app/routes.py]: Yeah bit lazy of me. Fixed.

File: test/test_app.py (hunk 2 above)
Comment (tunetheweb) [reply test/test_app.py]: Even more lazy of me! Proper test included now.

File: src/app/routes.py [hunk 3]
```diff
@@ -45,12 +45,20 @@ def fluff_results():
 
     dialect = request.args["dialect"]
     linted = lint(sql, dialect=dialect)
+    fixed_sql = ""
+
+    # Fix can fail badly if linting fails so wrap in a try/except.
+    # See: https://github.com/sqlfluff/sqlfluff-online/issues/25
+    try:
+        fixed_sql = fix(sql, dialect=dialect)
+    except:
```
Comment (nolanbconaway) [inline src/app/routes.py:54]: Nit: do we know anything more about the kinds of exceptions that might be raised? Minimally an `Exception`?

File: src/app/templates/index.html [hunk 4]
```diff
@@ -53,7 +53,7 @@ <h2>SQLFluff's Fixed SQL</h2>
         <tr>
             <th scope="row">{{violation['code']}}</th>
             <td>{{violation['line_no']}} / {{violation['line_pos']}}</td>
-            <td>{{violation['description']}}</td>
+            <td><pre>{{violation['description'].replace('\\n','\n')}}</pre></td>
```
Comment (nolanbconaway) [inline src/app/templates/index.html:56]: Verrrrry slick! Can you think of a way to get this tested or is this one of those things that's not worth testing in our opinion? I am a maniac for testing

**Review by nolanbconaway — APPROVED** [review]
LGTM- just a few nits in there for you to fix if you also feel the itch

File: src/app/routes.py (hunk 3 above)
Comment (tunetheweb) [reply src/app/routes.py]: Any Exception here is a problem. Added `Exception`.

Do wonder why `fix` raises this error as an exception but `lint` just as an unparseable error? But that's more a problem for sqlfluff, than sqlfluff-online and maybe there's a good reason.

File: src/app/templates/index.html (hunk 4 above)
Comment (tunetheweb) [reply src/app/templates/index.html:56]: Added a test case.

Also added coverage and another test to ensure 100% coverage coz I'm a manic for 100% coverage 😀

**Review by nolanbconaway — APPROVED** [review]
Nice! thx for 💀  all of these issues!

**Comment by nolanbconaway** [comment]
I'm happy to merge but will wait a bit in case you wanted to do it!

**Comment by tunetheweb** [comment]
Not bothered one way or another! 😁

But did change the repo to squash on merge as per discussion on the main repo. So will merge that now.

**Pull request merged**